core/payment-driver/dummy/src/lib.rs

```diff
@@ -54,65 +57,40 @@ impl PaymentDriver for DummyDriver {
     fn get_account_balance<'a>(
         &'a self,
         _address: &str,
-    ) -> Pin<Box<dyn Future<Output = Result<AccountBalance, PaymentDriverError>> + 'static>> {
+    ) -> Pin<Box<dyn Future<Output = Result<BigDecimal, PaymentDriverError>> + 'static>> {
         let amount = "1000000000000000000000000";
-        Box::pin(future::ready(Ok(AccountBalance {
-            base_currency: Balance {
-                currency: Currency::Gnt,
-                amount: utils::str_to_big_dec(&amount).unwrap(),
-            },
-            gas: Some(Balance {
-                currency: Currency::Eth,
-                amount: utils::str_to_big_dec(&amount).unwrap(),
-            }),
-        })))
+        Box::pin(future::ready(Ok(utils::str_to_big_dec(&amount).unwrap())))
     }
 
     fn schedule_payment<'a>(
         &self,
-        invoice_id: &str,
-        amount: PaymentAmount,
+        amount: BigDecimal,
         sender: &str,
         recipient: &str,
         _due_date: DateTime<Utc>,
-    ) -> Pin<Box<dyn Future<Output = Result<(), PaymentDriverError>> + 'a>> {
+    ) -> Pin<Box<dyn Future<Output = Result<String, PaymentDriverError>> + 'a>> {
         let payments = self.payments.clone();
         let details = PaymentDetails {
             recipient: recipient.to_string(),
             sender: sender.to_string(),
-            amount: amount.base_currency_amount,
+            amount: amount,
             date: Some(Utc::now()),
         };
-        let invoice_id = invoice_id.to_string();
+        // FIXME: generate payment_order_id?
+        let payment_order_id = "123".to_string();
 
         Box::pin(async move {
-            match payments.lock().await.entry(invoice_id.clone()) {
-                Entry::Occupied(_) => Err(PaymentDriverError::PaymentAlreadyScheduled(invoice_id)),
+            match payments.lock().await.entry(payment_order_id.clone()) {
+                // FIXME: need other properties to check if Payment is already Scheduled
```

Comment:
Depending on the question above, how can we ensure payments are only scheduled once? Or is this no longer the driver repsonsibility

Reply:
Not the driver's responsibility. Twice ordered, twice served.